lib/tasks/widgets/new_task_dialog.dart

```diff
@@ -2,20 +2,29 @@ import 'package:flutter/material.dart';
 import 'package:studypoints/tasks/data/task.dart';
 
 class NewTaskDialog extends StatefulWidget {
+  final Task task;
+  NewTaskDialog({this.task});
+
   @override
   _NewTaskDialogState createState() => _NewTaskDialogState();
 }
 
 class _NewTaskDialogState extends State<NewTaskDialog> {
   int priority = 0;
   DateTime dueDate;
-  TextEditingController titleController = TextEditingController();
+  TextEditingController titleController; // = TextEditingController();
   List<TextEditingController> subtaskControllers = [];
   List<Widget> subtaskInputs = [];
   final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
 
   @override
   Widget build(BuildContext context) {
+    if (titleController == null)
+      titleController = TextEditingController(text: widget?.task?.title);
```

Comment:
Here would be a good place to use the `??=` assignment operator. Also, `widget` should not be `null`, so you should not have to check for that.